Dockerfile

```diff
@@ -0,0 +1,54 @@
+FROM debian:stretch-20180213
+
+# Install all dependencies from manifest.scm
+RUN apt-get update && apt-get install -y \
+    autoconf \
+    automake \
+    bash \
+    bison \
+    cmake \
+    coreutils \
+    curl \
+    device-tree-compiler \
+    diffutils \
+    doxygen \
+    expat \
+    file \
+    findutils \
+    flex \
+    gawk \
+    gcc-6 \
+    g++ \
+    git \
+    libgmp-dev \
+    make \
+    grep \
+    gzip \
+    libedit-dev \
+    libssh2-1-dev \
+    libtool \
+    libusb-1.0-0-dev \
+    libxml2-dev \
+    mpc \
+    libmpfr-dev \
+    ninja-build \
+    libncurses-dev \
+    libssl-dev \
+    patchelf \
+    pkg-config \
+    python2.7 \
+    sed \
+    swig \
+    tar \
+    texinfo \
+    zlib1g-dev \
+    zsh
+
+RUN ln -s /usr/bin/python2.7 /usr/bin/python2
+RUN git clone https://github.com/riscv-rust/riscv-rust-toolchain.git
+
+WORKDIR /riscv-rust-toolchain
+RUN git submodule update --init --recursive
+
+ENV PATH=/riscv-rust-toolchain/toolchain/bin/:$PATH:/riscv-rust-toolchain/build/x86_64-unknown-linux-gnu/stage0/bin
```

Comment:
This is the only part of it that is a little bit weird. I had to add:

```
/riscv-rust-toolchain/build/x86_64-unknown-linux-gnu/stage0/bin
```

to `$PATH` in order for `make xargo` to work. Otherwise it didn't seem to build any other version of cargo?

Reply:
Did you have any problems with openocd from within docker?

Reply:
Not yet. I haven't actually tried doing anything other than compiling the toolchain, though